### ISSUE TITLE
User Expectation - Copy-Item when called with the Force parameter should copy the item to the full destination path provided even if part of that path doesn't exist

### ISSUE THREAD

**Issue by kilasuit** [PowerShell/PowerShell#19656]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

`Copy-Item C:\tmp\file.txt -Destination C:\tmp\folder\file.txt -Force`

### Expected behavior

```console
Folder and file is created in provided location as the Force parameter has been applied
```

### Actual behavior

```console
errors with Copy-Item: Could not find a part of the path
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.0-preview.3
PSEdition                      Core
GitCommitId                    7.4.0-preview.3
OS                             Microsoft Windows 10.0.25357
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jhoneill** [comment]
@kilasuit  "expection"  seems a cross between exception and expectation, and it might be better to say what the expectation is in the title 

If I do something like    
`copy *.jpg    \pictures\2023\may `
if 2023 exists and may is not a pre-existing directory it will copy each of the files over the last with the name "may"
If 2023 doesn't exist should it create 2023 and name copy each file over "may", or 
should it create 2023 and a may subdirectory and copy the files properly? 
  
I think it errors out because it can't reliably determine the users intent.

**Comment by kilasuit** [comment]
Was meant to be expectation, whoops.
Will update when I get back to computer.

**Comment by kilasuit** [comment]
@237dmitry thanks for the suggest as I do understand that New-Item could have been used here as per your snippet as well as other ways like Robocopy too. 
The repro I provided was based on so that I didn't forget to raise this whilst I was doing other things and would be the most natural & logical way of thinking of copying a file to a provided path.

@jhoneill the particular user story in my mind for the repro I provided would be, 
> User wants to copy 1 specific item, in this case a file to a destination, keeping the base name of the item as is in the destination, without having to pre-create the rest of the path where the new item and it's content will be copied to, much like how tools like robocopy can do this for 1/many files already

There are also likely many other user stories with the Copy-Item cmdlet that either have been missed or could be revisited to make this cmdlet function more intelligently & in line with user expectations in future. This was just a specific that annoys me everytime when I come across it.
